lib/reqTemplating.js

```diff
@@ -0,0 +1,52 @@
+'use strict';
+
+var yaml = require('js-yaml'),
+    fs = require('fs'),
+    preq = require('preq'),
+    template = require('url-template');
+
+function _assemble(spec) {
+    var traverse = function(path, key) {
+        // A simple dotted path reference
+        var code = "req"
+        path.split(/\./).forEach(function(pathBit, index) {
+            if (index) {
+                code += "['"+encodeURIComponent(pathBit)+"']";
+            }
+        });
+        return "newReq['"+key+"'] = "+code+";";
```

Comment:
It needs to be to make sure that it's valid JS. The simplest way to do this is `JSON.stringify('foo"bar') --> "foo\"bar"`